### ISSUE TITLE
[FR]: Add a Swift Package Plugin for Crashlytics DSYM uploads

### ISSUE THREAD

**Issue by BrentMifsud** [firebase/firebase-ios-sdk#11992]
### Description

With Swift package plugin support, now widely available, it would be nice to have the Crashlytics DSYM upload script available as a package plugin.

Today we have to setup a build phase for this. 

With a package plugin, the setup would be integrated into Xcode. Making onboarding much easier for new users.

### API Proposal

n/a

### Firebase Product(s)

Crashlytics

**Comment by samedson** [comment]
Hey @BrentMifsud - thanks for the suggestion. Was doing some digging and from https://github.com/apple/swift-package-manager/blob/main/Documentation/Plugins.md, it looks like Package Plugins will be restricted from accessing the network.

This would prevent upload-symbols from sending symbols to Crashlytics. I need to do some digging to see if there's a way to get the user's permission, but if someone has information about this it would be helpful for our exploration.

<img width="948" alt="Screenshot 2023-11-22 at 2 06 00 PM" src="https://github.com/firebase/firebase-ios-sdk/assets/555046/3a61b71e-7354-4ef8-8fae-2409084336ec">

**Comment by BrentMifsud** [comment]
Ah that's unfortunate. I didn't realize they put network limitations on package plugins.

**Comment by samedson** [comment]
Yeah, was definitely excited about this feature for this use case. But I guess for now I'll leave this open in case anyone finds something out that unblocks this.

**Comment by SwiftNativeDeveloper** [comment]
Has anyone opened up a feedback to Apple on this to allow plugins to talk to the rest of the world?
